sphinx_click/ext.py

```diff
@@ -273,15 +273,15 @@ def _generate_nodes(self, name, command, parent=None, show_nested=False):
 
         # Title
 
+        source_name = ctx.command_path
```

Comment:
I don't think we should default to this for every user. For some particularly large applications, with lots of nesting, this would be unnecessarily verbose. Could you add a configuration option to enable/disable this? You can do so by adding something like the following in the `setup` function at the bottom of this file:

    app.add_config_value('click_use_full_names', False, 'env', [bool])

You can then access this from `self.app.config.click_use_full_names` or similar (I don't remember exactly where)